extensions/keycloak-admin-rest-client/runtime/src/main/java/io/quarkus/keycloak/admin/client/reactive/runtime/ResteasyReactiveClientProvider.java

```diff
@@ -94,6 +96,12 @@ private ClientBuilderImpl registerJacksonProviders(ClientBuilderImpl clientBuild
                 }
             } else {
                 ObjectMapper newObjectMapper = new ObjectMapper();
+
+                // Same like JSONSerialization class. Makes it possible to use admin-client against older versions of Keycloak server where the properties on representations might be different
+                newObjectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+                // The client must work with the newer versions of Keycloak server, which might contain the JSON fields not yet known by the client. So unknown fields will be ignored.
+                newObjectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
```

Comment:
This change is interesting because it will only impact scenario when user customized objected mapper (or provided his own instance). I kinda expect that most users won't be impacted by this change. @geoand please review this file. Thank you

Reply:
These changes makes Admin client to not fail when working with older KC versions. I trust that @pskopek determined that KC admin client is always backwards compatible and it is fine that KC Admin client will be used with older versions.


But - these changes will only affect scenarios when user customized ObjectMapper bean or provided his own bean:

- https://quarkus.io/guides/all-config#quarkus-jackson_quarkus-jackson-fail-on-unknown-properties is set to `false` by default
- https://quarkus.io/guides/all-config#quarkus-jackson_quarkus-jackson-serialization-inclusion is not set by default

And what I say:

- either I read situation wrong
- or this needs to be done for all the scenarios, because I expect that affected scenario here is only minor, most users don't need to customize default ObjectMapper?

Please @geoand or @pskopek check again or prove me wrong. Maybe I misread situation. Thank you

Reply:
> This change is interesting because it will only impact scenario when user customized objected mapper (or provided his own instance). I kinda expect that most users won't be impacted by this change. @geoand please review this file. Thank you

Agreed

Reply:
I don't know what Keycloak's requirements are, but from the looks of this PR, it seems Keycloak needs a completely separate `ObjecMapper`

Reply:
Thanks for having a look @geoand

Reply:
Thanks for bringing it to my attention!

Reply:
@pskopek Can you please comment here ?

Reply:
Yes, Keycloak Admin needs differently configured `ObjectMapper`. I modified the PR, but I am not the Quarkus expert. @geoand , can you review the change, please?

Reply:
@geoand ping

Reply:
LGTM

Reply:
I am trying to say that this change is only done for what I consider less frequent scenario and will not affect most users. In which case it feels very random, why should it behave differently for some subset of users? Happy to be proved wrong, but I keep repeating myself and see no arguments.

Reply:
Oh, I hadn't read the entire context. I agree, it seems like your initial concern has still not been addressed

Reply:
Fwiw i turn on fail on unknown properties to spot missing data in requests done using http client requests. 

It would be unfortunate if that made me it impossible to use keycloack client but also can't be ok keycloak client modifies users serializer. 

Shouldn't it be using it's own serializer mapper?

Reply:
> It would be unfortunate if that made me it impossible to use keycloack client but also can't be ok keycloak client modifies users serializer.

we only reuse `ObjectMapper` when in a default setting, so when you turn something on, we stop using it.

> Shouldn't it be using it's own serializer mapper?

IIRC it was more efficient this way, but since the stand of @pskopek seems to be that it is safe to use this client version against older Keycloak version ( which makes me nervous), I can we will need to use own serializer mapper. I think changes would be just to drop code active when `canReuseObjectMapper` returns `true`. @pskopek can you do that (unless someone shots NO :-)) or do you require help?